service/app.js

```diff
@@ -0,0 +1,224 @@
+import cors from 'cors'
+
+import { Deck } from './models/Deck.js'
+import { User } from './models/User.js'
+
+const app = express()
+const port = 8000
+
+// Connect to MongoDB
+
+const connectionString = `mongodb+srv://${process.env.DB_USER}:${process.env.DB_PASSWORD}@cluster0.0y9om.mongodb.net`
+try {
+  await mongoose.connect(connectionString)
+} catch (err) {
+  console.log('error ', err)
+}
+
+// Middleware
+
+const exampleMiddleware = (req, res, next) => {
+  console.log('example middleware')
+  next()
+}
+
+app.use(cors())
+app.use(express.json())
+app.use(exampleMiddleware)
+
+// Routes
+
+// Get an individual card by id
+
+app.get('/cards/:id', async (req, res) => {
+  const card = await Deck.findOne({'cards._id': req.params.id})
+  if (card){
+    res.status(200).send(card.cards.slice(0,1))
+  } else {
+    res.sendStatus(404)
+  }
+})
+
+// Get all cards for a deck, with the option to paginate results
+
+app.get('/decks/:id/cards', async (req, res) => {
+  const limit = req.query.limit
+  const deck = await Deck.findById(req.params.id)
+  if (deck) {
+    res.status(200).send(deck.cards.slice(0, 5))
+  } else {
+    res.sendStatus(404)
+  }
+})
+
+// Get a deck by id
+
+app.get('/decks/:id', async (req, res) => {
+  const deck = await Deck.findById(req.params.id)
+  if(deck) {
+  res.status(200).send(deck)
+  } else {
+    res.sendStatus(404)
+  }
+})
+
+// Get a deck by user
+
+app.get('/users/:id', async (req, res) => {
+  const user = await Deck.findOne({'userId' : req.params.id})
+  if (user) {
+    res.status(200).send(user)
+  } else {
+    res.sendStatus(404)
+  }
+})
+
+// Create a deck
+
+app.post('/decks', (req, res) => {
+  const deckRequest = req.body
+  console.log('request body ', deckRequest)
+  const deck = new Deck ({
+    name: deckRequest.name,
+    size: deckRequest.size,
+    userId: deckRequest.userId,
+    cards: deckRequest.cards
+  })
+  deck.save()
+})
+
+// Create a card
+
+app.post('/cards', async (req, res) => {
+  const cardRequest = req.body
+  console.log('request body ', cardRequest)
+  if (cardRequest.deckId) {
+    const deck = await Deck.findById(cardRequest.deckId)
+    if (deck) {
+      deck.cards.push({
+        frontImage: cardRequest.frontImage,
+        frontText: cardRequest.frontText,
+        backImage: cardRequest.backImage,
+        backText: cardRequest.backText
+      })
+      deck.save()
+    }
+  } else {
+  res.sendStatus(502)
+  }
+})
+
+// Create a user
+
+app.post('/users', async (req, res) => {
+  const userRequest = req.body
+  console.log('request body ', userRequest)
+  const user = new User ({
+    displayName: userRequest.displayName,
+    email: userRequest.email,
+    joined: userRequest.joined,
+    active: userRequest.active,
+  })
+  user.save()
+})
+
+// Update a card
+
+app.put('/cards/:id', async (req, res) => {
+  const cardRequest = req.body
+  console.log('request body ', cardRequest)
+  if (cardRequest.deckId) {
+    const card = Deck.findById(req.params.id)
```

Comment:
What is being stored in 'card' here? What do we expect to be passed in as an Id?